hera_cal/smooth_cal.py

```diff
@@ -68,7 +68,7 @@ def single_iterative_fft_dly(gains, wgts, freqs, conv_crit=1e-5, maxiter=100):
     return np.sum(taus)
 
 
-def dpss_filters(freqs, times, freq_scale=10, time_scale=1800, eigenval_cutoff=1e-8):
+def dpss_filters(freqs, times, freq_scale=10, time_scale=1800, eigenval_cutoff=1e-9, Nmax=2000):
```

Comment:
How did you pick this default?

Reply:
Also, I'd suggest piping this parameter through to the command line argparser.

Reply:
That was mostly chosen for performance reasons. After about N > 2000, calculating the DPSS vectors starts to take a while if you're generating the full set of N DPSS vectors. Also, as you start going to higher N, the approximation for the number of non-zero eigenvalues, Nz = 2 * N * fw * dt, gets fairly accurate. It's a somewhat arbitrary choice, but it does seem to be a good transition point between runtime and the accuracy of the estimation of the number of eigenfunctions we want to keep.

Reply:
Gotcha. I'd make a comment in the docstring about why that's the default and what happens if the user raises/lowers it. Since you don't think the user is likely to change it (and it doesn't change the final answer, only the runtime, correct?) perhaps it doesn't need to be piped through to the command line.

Reply:
I'm considering completely removing it as a parameter and using this speed-up as the default. It improves the runtime by quite a bit and the results end up being the same regardless of how many data points are used.

Reply:
As we discussed, magic numbers in the code aren't great. It's better to have it as a parameter, provide a reasonable default value, and warn the user in the docstring that changing the value is only for advanced users.

Reply:
Sorry, I meant that I'm considering completely removing `Nmax` from this function. The code within the `if` and `else` blocks end up giving the same results. The only difference between the two is that there is a minor indexing error in `uvtools.dspec.dpss_operator` that I'm going to put an issue in for. Here's a comparison between the two methods.

<img width="1009" alt="Screen Shot 2022-04-21 at 12 25 42 PM" src="https://user-images.githubusercontent.com/17678594/164537924-e783558b-6f61-4680-a844-8d8682358a73.png">